en/docs/how-tos/analytics/elk-based-analytics.md

```diff
@@ -0,0 +1,362 @@
+# ELK Based Analytics
+
+!!! note
+    Refer to the [compatibility documentation]({{base_path}}/faqs/#compatibility) to confirm the supported update level for this feature.
+
+![Deployment diagram]({{base_path}}/assets/img/how-tos/analytics/elk-based-analytics/architecture.png)
+
+### Analytics Data flow
+
+The new On-Premise Analytics solution for WSO2 API Microgateway will publish analytics data into a log file and that file will be used as the source for the analytics solution.
+
+ELK based WSO2 API Microgateway On-Premise Analytics deployment architecture has 4 main components.
+
+1. Filebeats
+2. Logstash
+3. Elasticsearch
+4. Kibana
+
+This section will cover the steps required to configure the WSO2 API-M and then publish it to an external ELK cluster.
+
+!!! note
+    The solution is developed on ELK 8.1.0 and will only be supported in Kibana versions 8.0.0 and higher.
+
+### Step 1 - Configuring API Microgateway
+
+#### Step 1.1 - Configuring the micro-gw.conf file.
+
+Open the &lt;MICRO\_GW\_HOME&gt;/conf/micro-gw.conf file and add the following configuration.
+
+``` toml
+[analytics.elkAnalytics]
+  enable = true
+```
+
+#### Step 1.2 - Enabling Logs
+
+!!! note
+    The LOG4J configuration file `log4j2.properties` is not available in the API Microgateway runtime by default since the logging is carried out by a in-built configuration factory. In order to configure API Microgateway runtime for ELK analytics, create a `log4j2.properties` configuration file in the location &lt;MICRO\_GW\_HOME&gt;/conf/.
```

Comment:
```suggestion
    The LOG4J configuration file `log4j2.properties` is not available in the API Microgateway runtime by default since the logging is carried out by an in-built configuration factory. In order to configure API Microgateway runtime for ELK analytics, create a `log4j2.properties` configuration file in the location &lt;MICRO\_GW\_HOME&gt;/conf/.
```

Reply:
Done with https://github.com/wso2/docs-mg/pull/184/commits/50d67d45206cb74d0620e3be7f91ee74c6d8bb38